src/ripple/protocol/Book.h

```diff
@@ -65,28 +65,16 @@ hash_append(Hasher& h, Book const& b)
 Book
 reversed(Book const& book);
 
-/** Ordered comparison. */
-int
-compare(Book const& lhs, Book const& rhs);
-
 /** Equality comparison. */
 /** @{ */
-bool
+[[nodiscard]] bool
 operator==(Book const& lhs, Book const& rhs);
-bool
-operator!=(Book const& lhs, Book const& rhs);
 /** @} */
 
 /** Strict weak ordering. */
 /** @{ */
-bool
-operator<(Book const& lhs, Book const& rhs);
-bool
-operator>(Book const& lhs, Book const& rhs);
-bool
-operator>=(Book const& lhs, Book const& rhs);
-bool
-operator<=(Book const& lhs, Book const& rhs);
+[[nodiscard]] std::weak_ordering
+operator<=>(Book const& lhs, Book const& rhs);
```

Comment:
I think you can just use defaulted comparisons for these. Simply add the following inside the declaration of `Book` as a `public` and you automatically get all six relational operators:

`auto operator<=>(Bool const&) const = default;`

**NOTE, HOWEVER, THAT THIS HAS A RISK:*** first, the order of the member variables matters (since the default comparison goes through each member in order) and if any additional variables are added, it might cause subtle breakage of comparisons.

With that in mind, I'm not sure it's worth defaulting, but just pointing it out.

Reply:
You are right. We could default here but the `operator<=>` becomes a bit fragile as you pointed out.

I would not default unless you insist. Please let me know.